### PR TITLE
Add configuration for Bash command line completion

### LINKED ISSUE #1: [feature] shell completion
Add completion on commands project names and tags may be useful.

### PR DESCRIPTION
**Pull request by SpotlightKid**
Fixes #1 

I've also added a new command `frames` to list all frame IDs to support completion for the `edit` and `remove` commands. I've put this into an extra commit, so you can choose to include it or not.

Also added some documentation for it in the readme.

### CODE REVIEW

File: watson.completion [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Bash completion suppoort for watson
```
Comment (willdurand) [inline watson.completion]: s/suppoort/suport

File: README.md [hunk 2]
```diff
@@ -52,6 +52,15 @@ $ cd Watson/
...
 ```
 
+### Command line completion
+
+If you use a Bash-compatible shell, you can install the `watson.completion`
+file from the source distribution as `/etc/bash.completion.d/watson` - or
+wherever your distribution keeps the Bash completion configuration files.
+After you restart your shell, you can then just type `watson` on your command
+line and then hit TAB to see all available commands. Depending on your input,
+it completes watson commands, command options, projects, tags and frame IDs
```
Comment (willdurand) [inline README.md]: Period is missing at the end of this sentence.

**Comment by willdurand** [comment]
Could you please pass [shellcheck](https://github.com/koalaman/shellcheck) on the script file?

**Comment by SpotlightKid** [comment]
I ran `shellcheck watson.completion`, but the complains it has do not make sense to me:

```
In watson.completion line 1:
# Bash completion support for watson
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang.

In watson.completion line 19:
          COMPREPLY=($(compgen -W "-e --edit" -- ${cur})) ;;
                                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
[a lot more of those...]
```

1. This is not a script, but a something to be sourced by `/etc/bashrc`, therefor it has no shebang line.
2. `${cur}` must not be quoted, because otherwise `COMPREPLY` is an array.

**Comment by willdurand** [comment]
Thank you! Looks good to me, so :+1:

**Comment by jmaupetit** [comment]
@SpotlightKid Thank you for this!

Commands and projects completions work just fine, but tags completion does not :( I always have `+0` as completion... 

FYI my bash release is `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)`

**Comment by SpotlightKid** [comment]
Hmm, maybe your version of `sed` works differently than mine. Look at line 33 or 55 where I use `sed` too add a plus sign in front of every line returned by `watson tags`, by using the placeholder `\0` for the entire regular expression match. Maybe your sed doesn't support this placeholder?

Can you replace `sed 's/.*/+\0/'` with `awk '$0="+"$0'` in both places and see if that works for you?

**Comment by willdurand** [comment]
@SpotlightKid don't you have to use a group for that? (i.e. adding parenthesis)

**Comment by SpotlightKid** [comment]
In fact, I just noticed that adding plus signs is not needed when completing the argument of the `-T` option to `log` and `report` (line 33). Amended the commit.

**Comment by SpotlightKid** [comment]
@willdurand `\0` normally is the entire match. It works on my system (`sed (GNU sed) 4.2.2`). However, the awk solution should work better, so I used this and squashed the commits again.

**Comment by willdurand** [comment]
cool! BTW, what's your OS? Because we run OSX here, and there are differences between the unix tools..

**Comment by SpotlightKid** [comment]
Manjaro (Arch Linux derivate)

**Comment by jmaupetit** [comment]
@SpotlightKid the `awk` solution works like a charm on OSX!

Thank you so much for this!

**Pull request merged**